UIforETW/Utility.cpp

```diff
@@ -19,7 +19,166 @@ limitations under the License.
 #include <fstream>
 #include <direct.h>
 
-std::vector<std::wstring> split(const std::wstring& s, char c)
+
+namespace {
+PCWSTR const WPAStartupFileName = L"\\Startup.wpaProfile";
+
+std::wstring GetDocumentsFolderPath()
+{
+	//must CoTaskMemFree when done!
+	PWSTR docsPathTemp = NULL;
+	const HRESULT docsPathResult = ::SHGetKnownFolderPath(FOLDERID_Documents, KF_FLAG_NO_ALIAS, NULL, &docsPathTemp);
+	if (FAILED(docsPathResult))
+	{
+		if (docsPathResult == E_INVALIDARG)

```

Comment:
You know, I was thinking that some computers might *not* have a Documents folder... but in retrospect, that's just plain silly. This is just pathological thoroughness. Removed.